### ISSUE TITLE
Distribute multiple Cloudflared releases (Previous and latest release)

### ISSUE THREAD

**Issue by xetorixik** [cloudflare/cloudflared#27]
As known it is only possible to download the latest version of Cloudflared
https://developers.cloudflare.com/argo-tunnel/downloads/

Please distribute multiple Cloudflared releases. 
Both the previous and the latest release.

So you can easily switch back to the previous release in case of problems like https://github.com/cloudflare/cloudflared/issues/23

**Comment by thoughtpolice** [comment]
FWIW, the `Makefile` indicates [here in the source code](https://github.com/cloudflare/cloudflared/blob/master/Makefile#L43) that the binary distributions are versioned, it's just that the URLs are not exposed. Out of curiosity, I just tried replacing the `-stable-` part of the URL with the actual version as shown there, and it worked.

e.g. the Argo Tunnel page lists

  https://bin.equinox.io/c/VdrWdbjqyF/cloudflared-stable-linux-amd64.deb

but you can get other versions via

  https://bin.equinox.io/c/VdrWdbjqyF/cloudflared-2018.10.3-linux-amd64.deb
  https://bin.equinox.io/c/VdrWdbjqyF/cloudflared-2018.10.0-linux-amd64.deb

**Comment by sssilver** [comment]
Yes @thoughtpolice that should work!

@zackbloom perhaps we need make this more obvious?

**Comment by donovan** [comment]
I wanted to try downgrading to 2019.2.0 to work around #75 so was pleased to find this issue. I was kinda surprised that old versions were displaying the same error as I was thinking it was due to 2019.2.1 upgrading to Go 1.11. then I noticed this:

```
# dpkg -i cloudflared-2018.11.0-linux-amd64.deb
Selecting previously unselected package cloudflared.
(Reading database ... 91403 files and directories currently installed.)
Preparing to unpack cloudflared-2018.11.0-linux-amd64.deb ...
Unpacking cloudflared (2019.2.1) ...
Setting up cloudflared (2019.2.1) ...
```
Looks like all the debs point at the latest one:

```
$ for rel in $(curl -s https://raw.githubusercontent.com/cloudflare/cloudflared/master/RELEASE_NOTES | grep ^201); do wget https://bin.equinox.io/c/VdrWdbjqyF/cloudflared-${rel}-linux-amd64.deb; done
$ sha256sum cloudflared*
ff013bce7fb5bef671c6491887963099b1fcceb75028a371c407d8ab1101c2ee  cloudflared-2018.10.0-linux-amd64.deb
ff013bce7fb5bef671c6491887963099b1fcceb75028a371c407d8ab1101c2ee  cloudflared-2018.10.1-linux-amd64.deb
ff013bce7fb5bef671c6491887963099b1fcceb75028a371c407d8ab1101c2ee  cloudflared-2018.10.2-linux-amd64.deb
ff013bce7fb5bef671c6491887963099b1fcceb75028a371c407d8ab1101c2ee  cloudflared-2018.10.3-linux-amd64.deb
ff013bce7fb5bef671c6491887963099b1fcceb75028a371c407d8ab1101c2ee  cloudflared-2018.10.4-linux-amd64.deb
ff013bce7fb5bef671c6491887963099b1fcceb75028a371c407d8ab1101c2ee  cloudflared-2018.10.5-linux-amd64.deb
ff013bce7fb5bef671c6491887963099b1fcceb75028a371c407d8ab1101c2ee  cloudflared-2018.11.0-linux-amd64.deb
ff013bce7fb5bef671c6491887963099b1fcceb75028a371c407d8ab1101c2ee  cloudflared-2018.12.0-linux-amd64.deb
ff013bce7fb5bef671c6491887963099b1fcceb75028a371c407d8ab1101c2ee  cloudflared-2018.12.1-linux-amd64.deb
ff013bce7fb5bef671c6491887963099b1fcceb75028a371c407d8ab1101c2ee  cloudflared-2018.8.0-linux-amd64.deb
ff013bce7fb5bef671c6491887963099b1fcceb75028a371c407d8ab1101c2ee  cloudflared-2019.1.0-linux-amd64.deb
ff013bce7fb5bef671c6491887963099b1fcceb75028a371c407d8ab1101c2ee  cloudflared-2019.2.0-linux-amd64.deb
ff013bce7fb5bef671c6491887963099b1fcceb75028a371c407d8ab1101c2ee  cloudflared-2019.2.1-linux-amd64.deb
```

**Comment by nicosmaris** [comment]
All the debs return "2019.2.1" when running `cloudflared --version`

**Comment by donovan** [comment]
Just to be clear, it would be nice if older versions were available. Having what look like older files available for download that are actually links to the latest file is a recipe for confusion.

Here are the states:

1. Older files are available (eg cloudflared-2018.12.1-linux-amd64.deb)
2. No older files are available (what the doanload page currently implies)
3. Older files "exist" but point at the latest file.

From my point of view 1 is best and 3 is worst. It looks like 3 is the current situation.

**Comment by TownLake** [comment]
Thanks for the feedback; we've started publishing release assets here on GitHub which include older releases.

https://github.com/cloudflare/cloudflared/releases
